controllers/azuremachine_reconciler.go

```diff
@@ -48,8 +47,11 @@ type azureMachineService struct {
 }
 
 // newAzureMachineService populates all the services based on input scope.
-func newAzureMachineService(machineScope *scope.MachineScope, clusterScope *scope.ClusterScope) *azureMachineService {
-	cache := resourceskus.NewCache(clusterScope, clusterScope.Location())
+func newAzureMachineService(machineScope *scope.MachineScope, clusterScope *scope.ClusterScope) (*azureMachineService, error) {
+	cache, err := resourceskus.GetCache(clusterScope, clusterScope.Location())
```

Comment:
note that this introduces a subtle edge case, where the cached sku data is not refresh. we could potentially mitigate with background refresh. also, the data inside a single sku cache currently refreshes on first use instead of instantiation which probably slows things down quite a bit (vs instantiating, kicking off refresh in goroutine, and then waiting for it to be ready before allowing the cache to beused).

Reply:
I've opted to make the LRU cache a TTL/LRU cache for the time being. Your solution with a go routine would probably be more efficient, but I didn't want to get too deep into it in this PR.

Reply:
I'll open a second PR with the Cache and refactor the `ClusterScope` dep.

Reply:
ClusterScope dep is still there right? Should I open an issue?

Reply:
Oops... fixup incoming.

Reply:
:sparkles:  Clean up! :sparkles: :